src/main/java/seedu/address/model/module/Grade.java

```diff
@@ -22,20 +22,38 @@
     public static final String GRADE_VALIDATION_REGEX =
             "A\\+|A\\-|A|B\\+|B\\-|B|C\\+|C|D\\+|D|F|CS|CU";
 
+
+    private static final String EMPTY_VALUE = "NIL";
+
+    /**
+     * State of the grade
+     */
+    public final GradeState gradeState;
+
     /**
      * Immutable grade value.
      */
     public final String value;
 
+    public Grade() {
```

Comment:
Yea I was stuck between this and that choice also, I didn't have much preference and just implemented this. I will change to yours since now it has '2' votes haha.